### PR TITLE
Fix parquet file reader.

### PR DESCRIPTION
**Pull request by hjarraya**
Previous parquet reader was taking a long time to parse date-time columns.

### CODE REVIEW

File: gordo/machine/dataset/data_provider/ncs_reader.py [hunk 1]
```diff
@@ -346,13 +346,16 @@ def read_tag_files(
                 if dry_run:
                     logger.info("Dry run only, returning empty frame early")
                     return pd.Series()
-
+                before_downloading = timeit.default_timer()
                 with adls_file_system_client.open(file_path, "rb") as f:
                     df = file_type.read_df(f)
                     df = df.rename(columns={"Value": tag.name})
+                    df.index = pd.to_datetime(df.index, utc=True)
```
Comment (epa095) [inline gordo/machine/dataset/data_provider/ncs_reader.py]: This looks a bit weird for the csv case now, with double converting

Comment (hjarraya) [reply gordo/machine/dataset/data_provider/ncs_reader.py]: I am not sure in which date format in the current parquet files in datalake, if you think this is a double conversion, I'd rather have it than not.

Comment (epa095) [reply gordo/machine/dataset/data_provider/ncs_reader.py]: The point is that the way it was before was that `file_type.read_df(f)` handles all conversion including time parsing. Now it still does that for the csv case (and then parses again here) while for the parquet it's only here. You should either handle the timeseries parsing inside `read_df(f)` for both the csv and parquet readers, or in none of them.

File: gordo/machine/dataset/data_provider/ncs_reader.py [hunk 2]
```diff
@@ -346,13 +346,16 @@ def read_tag_files(
...
                     df = df[~df["Status"].isin(remove_status_codes)]
                     all_years.append(df)
-                    logger.info(f"Done parsing file {file_path}")
+                    logger.info(
+                        f"Done parsing in {(timeit.default_timer()-before_downloading):.2f} seconds for file {file_path}"
```
Comment (flikka) [inline gordo/machine/dataset/data_provider/ncs_reader.py]: I suggest rewording this logging statement to clarify that it is both downloading and parsing that is being timed.

File: gordo/machine/dataset/data_provider/file_type.py [hunk 3]
```diff
@@ -100,10 +99,6 @@ def __init__(self, time_series_columns: TimeSeriesColumns):
     def read_df(self, f: BinaryIO) -> pd.DataFrame:
         columns = self.time_series_columns.columns
         datetime_column = self.time_series_columns.datetime_column
-        pf = pa.ParquetFile(f)
-        table = pf.read(columns)
-        df = table.to_pandas()
-        df[datetime_column] = df[datetime_column].apply(
-            lambda v: pd.to_datetime(v, utc=True)
```
Comment (flikka) [inline gordo/machine/dataset/data_provider/file_type.py:107]: As @epa095 points out, if you remove the pd.to_datetime from the read_df in the ParquetFileType, and put it into the read_df in ncs_reader instead, you should remove it from the CsvFileType. Alternatively, just add the pd.to_datetime statement to ParquetFileType again, and remove it from read_df.

**Review by flikka — CHANGES_REQUESTED** [review]
Two comments:
* The pd.to_datetime should be _either_ in the *FileType classes _or_ in the read_df. Not in "half of" the FileType classes AND in the read_df function, that is very confusing. 
* A slightly unrelated issue (perhaps a good idea to fix here, or in a separate issue): The Parquet files appears to _not_ be sorted on the timeindex. This will probably cause problems (either crashing, or even worse, silent logic catastropies) if the dataframes are _not_ resampled. Usually we do resample, and then I _think_ there is an automatic sorting happening.

File: gordo/machine/dataset/data_provider/file_type.py (hunk 3 above)
Comment (hjarraya) [reply gordo/machine/dataset/data_provider/file_type.py:107]: moved it to ParqutFileType.

File: gordo/machine/dataset/data_provider/ncs_reader.py (hunk 1 above)
Comment (hjarraya) [reply gordo/machine/dataset/data_provider/ncs_reader.py]: You have a good point, moved to ParqutFileType.

**Comment by hjarraya** [comment]
Added sort_index after reading each file.

**Review by flikka — APPROVED** [review]

**Comment by flikka** [comment]
Goodie 👍
